### ISSUE TITLE
Bug:

### ISSUE THREAD

**Issue by Calorion** [platers/obsidian-linter#423]
## Describe the Bug

Header increment does not recognize setext-style headings

## How to Reproduce

1. Create an Obsidian document with headers as below
2. Lint the file
3. "Header 3" is incorrectly changed to an H2

Example to reproduce issue with

```
Header 1
========

Header 2
--------

### Header 3

```

## Expected Behavior

```
Header 1
========

Header 2
--------

### Header 3
```

## Actual Behavior

```
Header 1
========

Header 2
--------

## Header 3
```

## Device

- [ ] Desktop
- [X] Mobile

**Comment by pjkaufman** [comment]
Thanks for reporting this. I don't think that this rule is designed for the style mentioned. Could you link to the style information for the set-text style headings as all headings the linter deals with are the atx/starting hashtag style?

**Comment by pjkaufman** [comment]
It looks like the equals sign right below the text is considered to be a part of tge heading. I guess we could swap how we identify headers, but that is a lot more extensive than what is mentioned here.

**Comment by pjkaufman** [comment]
A minus sign below text means an H2. It is unclear as to what the standard format for this is.

**Comment by Calorion** [comment]
https://daringfireball.net/projects/markdown/syntax#header

>Markdown supports two styles of headers, [Setext](http://docutils.sourceforge.net/mirror/setext.html) and [atx](http://www.aaronsw.com/2002/atx/).

>Setext-style headers are “underlined” using equal signs (for first-level headers) and dashes (for second-level headers). For example:

>This is an H1
>=============

>This is an H2
>-------------
>Any number of underlining =’s or -’s will work.

**Comment by pjkaufman** [comment]
Gotcha. I was aware of its existence, but was not aware of its actual rules. These rules predate when I started work on the plugin. I can check to see if anything can be done to fix this, but it may be something that is listed as not supported. However that will be determined by what info I have access to in the underlying libraries.

Thanks for clarifying that it is any number of `-` or `=` signs.

**Comment by pjkaufman** [comment]
Here is a question that would need to be answered before I could try to address this issue:
- What should happen when an H1 or H2 needs incrementing as the style will be ambiguous?
  -  I can retain the same style until it becomes an H3, but that means going from an H3 to an H2 would not be setext style
  - I could make this a drop down, but it is looking more and more like I should extract style settings to be something outside of rules and have them passed into rules as params to help allow consistency across different rules and reduce option duplication

**Comment by Calorion** [comment]
Hm. That’s a really good question. I think a setting in the Rule of whether to use setext or atx headers is the answer. Either way it should _recognize_ setext headers, but unless that option is set, it wouldn’t create them.

**Comment by pjkaufman** [comment]
Yeah it is something that would need to be added. Since this is something that needs to be addressed in all heading rules and not just this one rule, I am going to get to it once I get through the current list of features to work on. I will put it at the bottom of the queue for things to work on above the backlog content.
